src/Proxy.hs

```diff
@@ -372,6 +377,96 @@ runGRPC c k = do
         Nothing -> return yCookieMap
         Just scm -> return $ Map.union scm yCookieMap
 
+runGRPCV2 :: ClientMonad IO (GRPCResultV2 (FromProtoResult a))
+        -> (a -> Handler TypedContent)
+        -> Handler TypedContent
+runGRPCV2 = runGRPCV2WithNotFoundError Nothing
+
+-- |Run a GRPC request.
+runGRPCV2WithNotFoundError ::
+           Maybe ErrorMessage -- ^ Error message to include in the response on GRPC status code NOT_FOUND.
+        -> ClientMonad IO (GRPCResultV2 (FromProtoResult a)) 
+        -> (a -> Handler TypedContent)
+        -> Handler TypedContent
+runGRPCV2WithNotFoundError errM c k = do
```

Comment:
In hindsight, I believe this function should probably just propagate the GRPCResultV2 if it succeeded at the HTTP/2 layer.